docs/github-authentication.rst

```diff
@@ -0,0 +1,56 @@
+
+Setting Up A Template Repo
+--------------------------
+
+To create a new assignment, you need to
+
+1. Create the assignment within your classroom on GitHub and
+2. Create a template repository. This repo contains all of the files that the students will need to complete their homework.
+
+Step two can be implemented with abc-classroom by performing the following steps.
+
+
+1. Authenticate with GitHub
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+First, you need to authenticate with GitHub. This authentication step will
+allow you to automagically create the template repo within your chosen organization.
+
+Before you can authenticate, you need a few things already installed on your
+computer.
+
+1. Git must be set up locally on your machine.
+2. The ``abc-classroom`` package should be installed locally as well. If you don't have
+   ``abc-classroom`` installed, you can run the command
+
+   $ pip install git+https://github.com/earthlab/abc-classroom.git
+
+   to install it.
+3. The authentication requires a valid GitHub username and password.
+
+To authenticate, run:
+
+    $ abc-init
+
+.. note::
+   When you run ``abc-init``, you may get an error saying you have an outdated
```

Comment:
@nkorinek i'm curious about the errors that you saw here. can you provide any examples for me to better understand what is happening (just respond to this comment not in the doc). i'm trying to understand what it might need to add or update. for instance are our dependencies incorrect or out of date?

Reply:
@lwasser I only ran into an issue with this when I was trying to run abc-init in my base conda environment, and my papermill was not up to date enough. Once I updated papermill it ran. Not sure if this is relevant enough for people who would be installing it into the earth-analytics-python environment or the abc-dev environment. But thought I'd go on the safe side. We can take it out if need be.

Reply:
ok got it. thank you @nkorinek !! i'll look into this a bit more tomorrow.
thank you for all of the great PR's today!!